provider/resource_server.go

```diff
@@ -40,6 +43,49 @@ func resourceServer() *schema.Resource {
 				Elem:        resourceInterface(),
 				Description: "Private interface properties",
 			},
+			"os_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Description: "OS id, use the following command to select one " +
+					"`curl -s -H \"Authorization: Bearer $ONLINE_TOKEN\"  " +
+					"https://api.online.net/api/v1/server/operatingSystems/{server id}`",
+			},
+			"user_login": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User login",
+			},
+			"user_password": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "User password",
+			},
+			"root_password": &schema.Schema{
+				Type:        schema.TypeString,
+				Required:    true,
+				Description: "Root password",
+			},
+			"partitioning_template_ref": &schema.Schema{
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				Description: "UUID of the partitioning template created from " +
+					"https://console.online.net/fr/template/partition",
+			},
+			"ssh_keys": &schema.Schema{
```

Comment:
Should also be able to add these as a resource by calling `/user/key/ssh`
Could be fine not to have this on merging this PR, but a TODO should be added.